src/spikeinterface/qualitymetrics/quality_metric_calculator.py

```diff
@@ -91,6 +91,7 @@ def _run(self, verbose, **job_kwargs):
         progress_bar = job_kwargs["progress_bar"]
 
         unit_ids = self.sorting.unit_ids
+        self._warn_on_empty_units()
```

Comment:
I think here instead of warning we should compute only metrics on non empty and then will the final dataframe with nan.
what do you think ?

Reply:
Done!